component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -728,9 +728,15 @@ private boolean isValidTokenFederatedUser(int token, AuthenticationContext conte
 
         String secretKey = null;
         if (context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL) != null) {
+            Map<String, String> claimProperties = TOTPUtil.getClaimProperties(context.getTenantDomain(),
+                    TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL);
             try {
-                secretKey = TOTPUtil.decrypt(context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL).
-                        toString());
+                if (claimProperties.get("EnableEncryption") != null) {
```

Comment:
Shall we add the value check of "EnableEncryption" as an improvement in another cycle ?